### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by bertagh**

### CODE REVIEW

File: Assignment4/bertagh-051730600/task07.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
+# SPARQL
+print("SPARQL")
+
+q5 = prepareQuery('''
+SELECT ?Entity WHERE {
+  ?Entity foaf:knows ?Entity1.
+  OPTIONAL { ?Entity foaf:knows ?Entity2. FILTER (?Entity1 != ?Entity2) }
```
Comment (dgarijo) [inline Assignment4/bertagh-051730600/task07.py:156]: wy the optional

File: Assignment4/bertagh-051730600/task07.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+#RDFLib
+print('RDFLib')
+
+entities = set()
+
+for s,p,o in g.triples((None, FOAF.knows, None)):
+  for s1,p1,o1 in g.triples((s, FOAF.knows, None)):
+    if o1 != o and s1 not in entities:
+      print(s)
+      entities.add(s1)
```
Comment (dgarijo) [inline Assignment4/bertagh-051730600/task07.py:179]: This lists all entities who know someone, but it does not do what the exercise asks for

File: Assignment4/bertagh-051730600/task08.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+#We add the missing values to g1
+g1.add((ns.JohnDoe, VCARD.Given, Literal("John")))
+g1.add((ns.SaraJones, VCARD.Given, Literal("Sara")))
+g1.add((ns.SaraJones, VCARD.Family, Literal("Jones")))
+g1.add((ns.SaraJones, VCARD.EMAIL, Literal("sara.jones@data.org")))
+g1.add((ns.JohnSmith, VCARD.Family, Literal("Smith")))
+g1.add((ns.HarryPotter, VCARD.EMAIL, Literal("hpotter@hogwarts.org")))
```
Comment (dgarijo) [inline Assignment4/bertagh-051730600/task08.py:56]: ?? You must not add the knowledge by hand!!

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address remaining changes

**Comment by anaigmo** [comment]
Submitted and mostly fixed, no more PRs allowed

**Pull request merged**